app/src/main/java/com/example/hackernews/view/dialog/LoginDialog.kt

```diff
@@ -1,80 +1,66 @@
+import com.example.hackernews.factories.LoginViewModelFactory
 import com.example.hackernews.model.entities.User
-import com.example.hackernews.view.activities.MainActivity
-import com.example.hackernews.viewmodel.user.UserViewModel
+import com.example.hackernews.viewmodel.LoginViewModel
+import com.example.hackernews.viewmodel.Result
 import java.util.*
+import java.util.concurrent.Executors
 
-class LoginDialog(
-    private val userViewModel: UserViewModel,
-    val listener: LoginCallback,
-    val callContext: Context
-) : AppCompatDialogFragment() {
+class LoginDialog(private val activity: Activity) :
+    AppCompatDialogFragment() {
 
     private lateinit var binding: LoginLayoutBinding
+    private val loginViewModel: LoginViewModel by viewModels {
+        LoginViewModelFactory(requireContext())
+    }
 
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
         val builder = AlertDialog.Builder(activity)
         binding = LoginLayoutBinding.inflate(layoutInflater)
         initViews(builder)
-        binding.tvUsernameRequiredMessage.visibility = View.GONE
-        binding.tvPasswordRequiredMessage.visibility = View.GONE
-        builder.setPositiveButton("Register") { dialog, which ->
-            val username = Helper.trimEditText(binding.editUsername)
-            val password = Helper.trimEditText(binding.editPassword)
+        builder.setPositiveButton("Register") { _, _ ->
+            val username = binding.editUsername.text.toString()
+            val password = binding.editPassword.text.toString()
             registerUser(username, password)
         }
-        builder.setNegativeButton("Login") { dialog, which ->
-            val username = Helper.trimEditText(binding.editUsername)
-            val password = Helper.trimEditText(binding.editPassword)
-            val loggedUser = loginUser(username, password)
-            if(loggedUser != null) {
-                loggedUser.observe(this) {
-                    listener.onLoggedIn(username)
-                }
-            }
-            else {
-                listener.onLoggedFailed()
-                Log.e("DIALOG FAILED", "$loggedUser")
-            }
+        builder.setNegativeButton("Login") { _, _ ->
+            Toast.makeText(context, "Clicked on login", Toast.LENGTH_LONG).show()
         }
         return builder.create()
     }
 
-    private fun loginUser(username: String, password: String): LiveData<User?>? {
-        userViewModel.logUser(username, password)
-        return userViewModel.user
-    }
-
     private fun registerUser(username: String, password: String) {
-        val user: User = User(username, password, UUID.randomUUID().toString(), 0)
-        userViewModel.insert(user)
-        Toast.makeText(context, "Successfully registered $username", Toast.LENGTH_LONG).show()
+        val user = User(username, password, UUID.randomUUID().toString(), 1)
+        val executors = Executors.newSingleThreadExecutor()
```

Comment:
Great job with discovering Executors. It's much better to have thread switching in the repository itself. Move executors scheduling there.